modules/extensions/galasa-extensions-parent/dev.galasa.cps.etcd/src/main/java/dev/galasa/cps/etcd/internal/Etcd3DynamicStatusStore.java

```diff
@@ -127,24 +127,53 @@ public void put(@NotNull String key, @NotNull String value) throws DynamicStatus
      */
     @Override
     public void put(@NotNull Map<String, String> keyValues) throws DynamicStatusStoreException {
-        Txn txn = kvClient.txn();
         PutOption options = PutOption.DEFAULT;
+        setPropertiesIntoDSS(keyValues, options);
+    }
+
+    /**
+     * Put multiple key-value pairs into the DSS that expire after a given amount of time in seconds
+     * 
+     * @param keyValues the key-value pairs to be stored
+     * @param timeToLiveSecs the amount of time in seconds for the key-value pairs to be available for before expiring
+     * @throws DynamicStatusStoreException if there was an error accessing etcd
+     */
+    @Override
+    public void put(@NotNull Map<String, String> keyValues, @NotNull long timeToLiveSecs)
```

Comment:
Previously, we only had the ability to set a single property with a time-to-live. Since we need to set multiple properties into the DSS to cache a user, this has been added to allow us to set multiple properties at once with the same time-to-live.